esi_leap/api/controllers/v1/contract.py

```diff
@@ -102,19 +100,48 @@ def post(self, new_contract):
         return Contract(**c.to_dict())
 
     @wsme_pecan.wsexpose(Contract, wtypes.text)
-    def delete(self, contract_uuid):
+    def delete(self, contract_id):
         request = pecan.request.context
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:contract:delete', cdict, cdict)
 
-        c = contract.Contract.get(request, contract_uuid)
-        if c.project_id != request.project_id:
-            o = offer.Offer.get(request.project_id, c.offer_uuid)
-            if o.project_id != request.project_id:
+        permitted = ContractsController._contract_get_all_authorized_contracts(
+            contract_id, cdict)
+
+        permitted[0].cancel()
+
+    @staticmethod
+    def _contract_get_all_authorized_contracts(contract_id, cdict):
+
+        c_id = contract.Contract.get(contract_id)
```

Comment:
Maybe call it `contracts` instead of `c_id`?

Reply:
I just didn't like the fact that there is an imported module named contract, we are in a module named contract, the imported module uses a class called Contract and the module we are in defines a class called Contract.

Reply:
@mock.patch('esi_leap.api.controllers.v1.offer.offer.Offer.create')

Reply:
> I just didn't like the fact that there is an imported module named contract, we are in a module named contract, the imported module uses a class called Contract and the module we are in defines a class called Contract.

Ah, fair enough. However, if `contract.Contract.get` is returning multiple contract objects, the assigned variable should almost definitely be plural, and drop the `_id`. Maybe `contract_objs`?

Reply:
Fixed